mslib/msui/msui_mainwindow.py

```diff
@@ -468,6 +471,128 @@ def __init__(self, mscolab_data_dir=None, *args):
+
+    def operation_changed(self):
+        folder_path = MSUI_CONFIG_PATH
+        file_name = self.listOperationsMSC.currentItem()
+        file_path = os.path.join(folder_path, file_name.text() + ".txt")
+        if os.path.exists(file_path):
+            result = QtWidgets.QMessageBox.question(self, "Saved Layout Detected", "Would you like to restore it?",
+                                                    QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No, QtWidgets.QMessageBox.Yes)
+            if result == QtWidgets.QMessageBox.Yes:
+                self.restore_operation_data()
+        print("opertaion is changed")
+
+    def store_operation_data(self):
+        self.file_name = self.listOperationsMSC.currentItem()
+        if self.listOperationsMSC.currentItem() == None:
+            message_box = QtWidgets.QMessageBox()
+            message_box.setWindowTitle("Select Operation")
+            message_box.setText("Select a operation to save layout")
+            message_box.setStandardButtons(QtWidgets.QMessageBox.Ok)
+            message_box.exec_()
+            return
+        self.widget_info = {}
+        top_level_widgets = QtWidgets.QApplication.topLevelWidgets()
+        for widget in top_level_widgets:
+            if widget.objectName() in self.allowed_widget_names:
+                positions = []
+                if isinstance(widget, QtWidgets.QMainWindow):
+                    geometry = widget.geometry().getRect()
+                    screen_index = QtWidgets.QApplication.desktop().screenNumber(widget)
+                    is_minimized = widget.isMinimized()
+                    positions.append(geometry)
+                    positions.append(screen_index)
+                    positions.append(is_minimized)
+                child_info = {}
+                for child in widget.findChildren(QtWidgets.QWidget):
+                    if isinstance(child, QtWidgets.QComboBox):
+                        child_info[child.objectName()] = {"items": [child.itemText(i) for i in range(child.count())],
+                                                        "current_index": child.currentIndex()}
+                    elif isinstance(child, QtWidgets.QCheckBox):
+                        child_info[child.objectName()] = {"checked": child.isChecked()}
+
+                itemlist = [child_info, positions]
+                self.widget_info[widget.objectName()] = itemlist
+
+        folder_path = MSUI_CONFIG_PATH
+        file_path = os.path.join(folder_path, self.file_name.text() + ".txt")
+        json_data = json.dumps(self.widget_info, indent=4)
+
+        with open(file_path, 'w') as file:
+            file.write(json_data)
+            message_box = QtWidgets.QMessageBox()
+            message_box.setWindowTitle("Select Operation")
+            message_box.setText(f"Dictionary has been successfully stored in {file_path}")
+            message_box.setStandardButtons(QtWidgets.QMessageBox.Ok)
+            message_box.exec_()
```

Comment:
This doesn't have to be inside the `with open(...)` context manager and would semantically fit better outside, since the context manager only closes the file when leaving the `with` block.

Reply:
Also, I think this message box could just use one of the "standard" ones, which would reduce this to one line: `QtWidgets.QMessageBox.information`.